component/src/main/java/org/wso2/extension/siddhi/execution/streamingml/clustering/kmeans/KMeansIncrementalSPExtension.java

```diff
@@ -0,0 +1,305 @@
+                                "points we will receive coordinates along each axis.",
+                        type = {DataType.DOUBLE} //how to give multiple types?
+                )
+
+        },
+        returnAttributes = {
+                @ReturnAttribute(
+                        name = "euclideanDistanceToClosestCentroid",
+                        description = "Represents the Euclidean distance between the current data point and the " +
+                                "closest centroid.",
+                        type = {DataType.DOUBLE}
+                ),
+                @ReturnAttribute(
+                        name = "closestCentroidCoordinatei",
+                        description = "This is a variable length attribute. Depending on the dimensionality(d) " +
+                                "we will return closestCentroidCoordinate1 to closestCentroidCoordinated",
+                        type = {DataType.DOUBLE}
+                )
+        },
+        examples = {
+                @Example(
+                        syntax = "from InputStream#streamingML:KMeansIncremental(modelName, numberOfClusters, " +
+                                "coordinateValue1, coordinateValue2)\"\n" +
+                                "select coordinateValue1, coordinateValue2, euclideanDistanceToClosestCentroid, " +
+                                "closestCentroidCoordinate1, closestCentroidCoordinate2\"\n" +
+                                "insert into OutputStream",
+                        description = "modelName='model1', numberOfClusters=2, This will select first two " +
+                                "distinct values in " +
+                                "events as initial model and predict and update the model for every event"
+                ),
+        }
+)
+public class KMeansIncrementalSPExtension extends StreamProcessor {
+    private float decayRate;
+
+
+    private int numberOfEventsReceived;
+    private int coordinateStartIndex;
+    private LinkedList<DataPoint> dataPointsArray;
+    private double[] coordinateValuesOfCurrentDataPoint;
+
+    private boolean modelTrained = false;
+    private Clusterer clusterer;
+    private int dimensionality;
+    private static final Logger logger = Logger.getLogger(KMeansIncrementalSPExtension.class.getName());
+
+    @Override
+    protected List<Attribute> init(AbstractDefinition abstractDefinition,
+                                   ExpressionExecutor[] attributeExpressionExecutors,
+                                   ConfigReader configReader, SiddhiAppContext siddhiAppContext) {
+        dataPointsArray = new LinkedList<>();
+        logger.setLevel(Level.ALL);
+
+        //expressionExecutors[0] --> modelName
+        if (!(attributeExpressionExecutors[0] instanceof ConstantExpressionExecutor)) {
+            throw new SiddhiAppValidationException("modelName has to be a constant.");
```

Comment:
This is not descriptive enough. Please check https://github.com/wso2-extensions/siddhi-execution-streamingml/blob/master/component/src/main/java/org/wso2/extension/siddhi/execution/streamingml/classification/perceptron/PerceptronClassifierUpdaterStreamProcessorExtension.java#L126 ..... Fix all the exceptions to be more descriptive.

Reply:
done